server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationMetaData.java

```diff
@@ -220,6 +228,97 @@ public CoordinationMetaData build() {
+        private String nodeId;
+        private String nodeName;
+
+        public VotingTombstone(DiscoveryNode node) {
+            this(node.getId(), node.getName());
+        }
+
+        public VotingTombstone(StreamInput in) throws IOException {
+            this.nodeId = in.readString();
+            this.nodeName = in.readString();
+        }
+
+        public VotingTombstone(String nodeId, String nodeName) {
+            this.nodeId = nodeId;
+            this.nodeName = nodeName;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(nodeId);
+            out.writeString(nodeName);
+        }
+
+        public String getNodeId() {
+            return nodeId;
+        }
+
+        public String getNodeName() {
+            return nodeName;
+        }
+
+        private static final ParseField NODE_ID_PARSE_FIELD = new ParseField("node_id");
+        private static final ParseField NODE_NAME_PARSE_FIELD = new ParseField("node_name");
+
+        private static String nodeId(Object[] nodeIdAndName) {
+            return (String) nodeIdAndName[0];
+        }
+
+        private static String nodeName(Object[] nodeIdAndName) {
+            return (String) nodeIdAndName[1];
+        }
+
+        private static final ConstructingObjectParser<VotingTombstone, Void> PARSER = new ConstructingObjectParser<>(
+                "voting_tombstone",
+                nodeIdAndName -> new VotingTombstone(nodeId(nodeIdAndName), nodeName(nodeIdAndName))
+        );
+
+        static {
+            PARSER.declareString(ConstructingObjectParser.constructorArg(), NODE_ID_PARSE_FIELD);
+            PARSER.declareString(ConstructingObjectParser.constructorArg(), NODE_NAME_PARSE_FIELD);
+        }
+
+        public static VotingTombstone fromXContent(XContentParser parser) throws IOException {
+            return PARSER.parse(parser, null);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            return builder.startObject()
+                    .field(NODE_ID_PARSE_FIELD.getPreferredName(), nodeId)
+                    .field(NODE_NAME_PARSE_FIELD.getPreferredName(), nodeName)
+                    .endObject();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            VotingTombstone that = (VotingTombstone) o;
+            return Objects.equals(nodeId, that.nodeId) &&
+                    Objects.equals(nodeName, that.nodeName);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(nodeId, nodeName);
+        }
+
+        @Override
+        public String toString() {
+            StringBuilder sb = new StringBuilder();
+            sb.append('{').append(nodeId).append('}');
```

Comment:
In `DiscoveryNode#toString()` the node name comes first (if present). I think we should do the same here.

Reply:
https://github.com/elastic/elasticsearch/pull/35832/commits/dbb21b2d1499a8c0edf58a6c36ec7ba023bbb862